src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1

```diff
@@ -215,4 +215,79 @@ class AgentPool: ADOSVTBase
         $agentPool = $null;
         return $controlResult
     }
+
+    hidden [ControlResult] CheckCredInEnvironmentVariables([ControlResult] $controlResult)
+    {
+        try 
+        {   
+            $agentPoolsURL = "https://dev.azure.com/{0}/{1}/_settings/agentqueues?queueId={2}&__rt=fps&__ver=2" -f $($this.SubscriptionContext.SubscriptionName), $this.ProjectId ,$this.AgentPoolId;
+            $agentPool = [WebRequestHelper]::InvokeGetWebRequest($agentPoolsURL);
+            $patterns = $this.ControlSettings.Patterns | where {$_.RegexCode -eq "SecretsInBuild"} | Select-Object -Property RegexList;
```

Comment:
Keep this logic only if $agentPool response object has some data

Reply:
As discussed with no need for this check